uniffi_aries_vcx/demo/app/src/main/java/org/hyperledger/ariesvcx/HomeScreen.kt

```diff
@@ -1,37 +1,76 @@
 package org.hyperledger.ariesvcx
 
+import android.util.Log
 import android.widget.Toast
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.Button
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.navigation.NavHostController
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.hyperledger.ariesvcx.utils.await
+
 
 @Composable
-fun HomeScreen(navController: NavHostController, setProfileHolder: (ProfileHolder) -> Unit, profileHolder: ProfileHolder?, connection: Connection?) {
+fun HomeScreen(
+    navController: NavHostController,
+    setProfileHolder: (ProfileHolder) -> Unit,
+    profileHolder: ProfileHolder?,
+    connection: Connection?,
+    walletConfig: WalletConfig,
+    connectionRequestState: Boolean,
+    httpClient: OkHttpClient
+) {
 
     val scope = rememberCoroutineScope()
     val context = LocalContext.current
 
-    val walletConfigState = WalletConfig(
-        walletName = "test_create_wallet_add_uuid_here",
-        walletKey = "8dvfYSt5d1taSd6yJdpjq4emkwsPDDLYxkNFysFD2cZY",
-        walletKeyDerivation = "RAW",
-        walletType = null,
-        storageConfig = null,
-        storageCredentials = null,
-        rekey = null,
-        rekeyDerivationMethod = null
-    )
+    var flagKeepFetching by remember {
+        mutableStateOf(true)
+    }
+
+    val request = Request.Builder()
+        .url("$BASE_RELAY_ENDPOINT/pop_user_message/${walletConfig.walletKey}")
```

Comment:
still an issue here with walletKey being used. should match the other string. you could also make it into some constant string;

`const val RELAY_USER_ID = "demo-user-1"